### PR TITLE
Release 0.52.0

### PR DESCRIPTION
**Pull request by qinqon**
Signed-off-by: Quique Llorente <ellorent@redhat.com>

**What this PR does / why we need it**:
Release 0.52.0

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by phoracek — COMMENTED** [review]
/lgtm
/approve

**Comment by RamLavi** [comment]
Hey @qinqon looks like restarting the `kubemacpool-cert-manager` is still causing issues, see [here](https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_cluster-network-addons-operator/821/pull-e2e-cluster-network-addons-operator-kubemacpool-functests/1383002783719886848) where 2 seemingly random tests failed..

From closer inspection, from some reason sometimes the cert pod does not change replicas to 0 while restarting..:
```
[1mSTEP[0m: updating deployment kubemacpool-cert-manager, pod replicas to be 0
[1mSTEP[0m: Waiting for expected ready pods to be 0
```

**Comment by qinqon** [comment]
Isnt't cnao going to reconcile that ? So it restore the realices while we set them to zero? Same for kmp.

**Comment by RamLavi** [comment]
> Isnt't cnao going to reconcile that ? So it restore the replicas while we set them to zero? Same for kmp.

This could be it, we need to find another way to restart the KMP pods..

**Comment by qinqon** [comment]
> > Isnt't cnao going to reconcile that ? So it restore the replicas while we set them to zero? Same for kmp.
> 
> This could be it, we need to find another way to restart the KMP pods..

Searching the pods and deleting them should be it.

**Comment by RamLavi** [comment]
> > > Isnt't cnao going to reconcile that ? So it restore the replicas while we set them to zero? Same for kmp.
> > 
> > 
> > This could be it, we need to find another way to restart the KMP pods..
> 
> Searching the pods and deleting them should be it.

IIRC, we used to do that, and changed it to restart by deployment [here](https://github.com/k8snetworkplumbingwg/kubemacpool/commit/c1a31b3c4491efffb71386889d602ad1f5313f3d)

I think we changed it to deployments from the same reason we want to change it now - because we are trying to change something (kill the pods) when we are not the only ones monitoring/controlling it, and that created annoying races.. 

so what do you think of this:
1. remember <old pod name>
2. "kill" pod by deployment.replicas-->0
3. wait until <old pod name> is gone
4. bring back pod by deployment.replicas-->1
5. wait until deployment.ready-replicas==1

basically the only change is in step 3... It's not pretty, but should eliminate races... What do you think @qinqon ?
